lib/jit_preloader.rb

```diff
@@ -25,6 +25,14 @@ def self.globally_enabled=(value)
     @enabled = value
   end
 
+  def self.query_dive_limit=(limit)
+    @dive_limit = [limit - 1, 1].max
+  end
+
+  def self.query_dive_limit
```

Comment:
thoughts on using attr_reader here?

Reply:
You'll need a class << self, or maybe a singleton_class.attr_reader

Reply:
As a second look, since globally_enabled is directly setting with value, I want to keep the pattern same.